In [15]:
##for 循环

n = int (input("input a number"))

def factor0(n):
    factor=1

    for i in range (1,n+1):
       
        factor=factor*i;
    
    return factor
print('The factor of n is:', factor0(n))


%timeit a=[factor0(n)]

input a number10
('The factor of n is:', 3628800)
100000 loops, best of 3: 2.72 µs per loop


In [13]:
## while 循环

n = int (input("input a number"))

def factor1(n):
    factor = 1
    i=1
    while i < n:
        factor=factor*i
        i=i+1
    return factor

print('The factor of n is:', factor1(n))

%timeit a=[factor1(n)]

input a number10
('The factor of n is:', 362880)
100000 loops, best of 3: 2.53 µs per loop


In [10]:
##递归
n=int(input("Input a number:" ))
def factor(i):
    if i == 1:
        return 1;
    else:
        return i * factor (i-1)

print (factor(n))

%timeit a=[factor(n)]


Input a number:10
3628800
100000 loops, best of 3: 2.84 µs per loop


我是在MacBook Pro OSX 版本10.11.6 环境下安装的Anaconda， 直接从官网上下载Anaconda最新安装包，然后一路安装就可以。
安装完成后再anaconda里面安装Spyder和ipython notebook，就可以直接打开编辑ipython环境。

主要问题有一个，就是如何在MacBook 下安装 不同版本的Python，如2.7 和 3.5 共存。
这里我从网上看到了一个方法，就是在terminal里安装最新的brew，用brew安装最新的pyenv包，通过pyenv包加载两个Python环境。网上有一个很详细的教程：http://www.cnblogs.com/blackmatrix/p/5591341.html 如果有类似问题可以看一下这个教程。最终完美解决了两个版本共存的问题。